packages/http/httpcall_common.js

```diff
@@ -5,9 +5,14 @@ makeErrorByStatus = function(statusCode, content) {
     return str.length > length ? str.slice(0, length) + '...' : str;
   };
 
+  // this solution is not browser compatible
+  // var contentToCheck = typeof Buffer != "undefined" && content instanceof Buffer ? content.toString() : content;
+  var contentToCheck = typeof content == "string" ? content : content.toString();
```

Comment:
Sometime we use encoding: null because of special encoding of the webpage (latin, ...). And when it happens you will have html content.
Maybe we can check if we found <html> node after we have stringified the Buffer. If it's ok, we can continue, if not, we should just return "failed [statusCode]"